Phase2-hgx185 Add 2 algorithms for HGCal to DD4Hep format

bsunanda opened the pull request:
#### PR description:

Convert 2  DDD algorithms for HGCal to dd4hep format

#### PR validation:

Matched the two algorithms. Final testing will be done while all algorithms of HGCal DDD will be transported

#### if this PR is a backport please specify the original PR:

It is intended for the master only

apsallid committed:
validation in mtv style - first steps

apsallid committed:
better keep the plots in a separate folder for now

apsallid committed:
Adding CaloParticleSelector and CaloParticle histogram skeleton.

apsallid committed:
Adding vertex related parameters in CaloParticleSelector

apsallid committed:
Adding some layer cluster validation plots

apsallid committed:
adding layer cluster plots and accessing differently rechit tools

apsallid committed:
forgot some variables

apsallid committed:
trying to make a stable version

apsallid committed:
Saving the HGCal-style seed of the 2d layer cluster

apsallid committed:
using max_element for the HGCal-style seed of the 2D cluster

jan-kaspar committed:
Eliminated uncessary BeamDivergenceVtxGenerator.h.

jan-kaspar committed:
Several small review comments addressed.

jan-kaspar committed:
useImprovedInitialEstimate made config parameter.

jan-kaspar committed:
Output labels made config parameters.

jan-kaspar committed:
Unnecessary type specifications removed.

jan-kaspar committed:
(Un)tracked mismatched fixed.

jan-kaspar committed:
(Un)tracked mismatch fixed.

jan-kaspar committed:
CTPPSProtonReconstruction renamed to CTPPSProtonProducer,
ctppsProtonReconstruction renamed to ctppsProtons.

jan-kaspar committed:
Run number set to match with 2016 range.

jan-kaspar committed:
Fit avoids using TF1 and TGraph.

jan-kaspar committed:
Run number corresponds to 2016 period.

jan-kaspar committed:
Use cm as length unit for optical functions.

jan-kaspar committed:
ctppsProtons in miniAOD.

apsallid committed:
Adding more histograms

apsallid committed:
Adding plotting support

apsallid committed:
Making a better html plotting structure and adding the missing accumulative material budget file

forthommel committed:
Replaced setters calls by a modified ProtonTrack constructor

forthommel committed:
Renaming of ProtonTrack to ForwardProton

forthommel committed:
Adapted RecoCTPPS to the naming change

forthommel committed:
Small fixes

forthommel committed:
Using single precision floats instead of doubles for vertex/momentum/covariance tracks attributes

forthommel committed:
Avoiding copies in proton collections filling

forthommel committed:
std::vector<edm::Ref<T> > → edm::RefVector<T>

forthommel committed:
CMS coding rules

forthommel committed:
Using the new Fwd definition of CTPPSLocalTrackLite

forthommel committed:
Added required streamers for edm::RefVector

jan-kaspar committed:
Implemented several suggestions for ProtonReco class.

jan-kaspar committed:
Scoring-plane positions given in cm.

forthommel committed:
Moved the .root optical functions container to cms-data repository

forthommel committed:
Matched light local tracks constructor to new signature

forthommel committed:
Corrected units for reconstruction validator

jan-kaspar committed:
Unit added in comment.

forthommel committed:
Smart pointers used whenever possible in validation package

jan-kaspar committed:
All inputs are checked to be valid.

jan-kaspar committed:
Added plots for SW tests and RelVals (lite tracks and proton reco).

jan-kaspar committed:
Alignment constants loaded from preparation DB.

jan-kaspar committed:
ESProducers from CondFormats/CTPPSReadoutObjects and Validation/CTPPS moved to CalibPPS/ESProducers.

jan-kaspar committed:
Eliminate dependence of RecoCTPPS on Validation.

jan-kaspar committed:
Updated CondFormats related to optical functions.

jan-kaspar committed:
Updated Cond records related to optical functions.

jan-kaspar committed:
Updated ESProducers related to optical functions.

jan-kaspar committed:
Updated EDProducers and analyzers related to optical functions.

jan-kaspar committed:
Updated configuration related to optical functions.

jan-kaspar committed:
Applied patch from code-checks.

forthommel committed:
Updated the optics file path according to https://github.com/cms-data/CalibPPS-ESProducers/pull/1

forthommel committed:
Modified the ESSource to deal with the diamond timing calibration file parsing

forthommel committed:
Example sqlite dumper amended

jan-kaspar committed:
CTPPSLHCInfoESSource.cc moved to CalibPPS/ESProducers.

jan-kaspar committed:
Removed obsolete configs.

jan-kaspar committed:
Added ESPrefer statements.

jan-kaspar committed:
Test configs updated.

apsallid committed:
commenting out densities for the moment

forthommel committed:
Added time offset/precision

apsallid committed:
plot tweaks

jan-kaspar committed:
IOV for optical functions can be specified.

jan-kaspar committed:
IOV temporarily hard-coded with both event-id and time specifications.

apsallid committed:
Merging with the work by Marco and Felice

rovere committed:
Fix PlotOnSide with separate option conflict

rovere committed:
Make separate plots browsable together

If Validation plots are produced using the 'separate' option, a static
web page will be created with the proper links to all the images. This
commit will add in that specific directory the index.php file, and all
its needed components, to be able to browse the content of the
directory automatically, hence visualizing all images at a glance.

rovere committed:
Order Layer number correctly

Layer numbering in the histogram has been modified to be correctly
padded with leading zeros, so that the plots will be correctly ordered
when viewed. Added also 2D plots using the PlotOnSideGroup feature.

I will come back on this to finalize the layer ordering in my side of the plots. It's not fully ok for the
moment

apsallid committed:
Avoiding hardcoded values in histograms

apsallid committed:
Changing the title of histograms to sensible layer numbering

apsallid committed:
Extending the remaining per layer histograms to the full range 0 to 103

apsallid committed:
Updating the post processor to include all plots and correcting the plotting index to go through all plots while giving a sensible layer numbering in z- and z+

forthommel committed:
Adapted to new json structure; cleanup of tests

forthommel committed:
Extra log cleanup

rovere committed:
Add Browse Folder link

rovere committed:
Improve style for few plots

rovere committed:
Fix layer number padding

apsallid committed:
plot tweaks

rovere committed:
Improve Score plots rendering

rovere committed:
Make energy weight choice more flexible

In order to test the behaviour of the scores using different energy
weights, the code has been generalized and the variable names
abstracted. Also, now the same threshold is used to associate a
CaloParticle to a LayerCluster and viceversa.

apsallid committed:
making different plots for z- and z+

jan-kaspar committed:
Bug fix: no input case covered.

rovere committed:
Improve style for cells plots

apsallid committed:
Correcting the title for efficiencies/duplicates/fakes/merges

jan-kaspar committed:
No access to track data when optics is not valid.

apsallid committed:
Updating the xbinlabels for efficiencies/duplicates/fakes/merges

apsallid committed:
Activating all plots

jan-kaspar committed:
Produce as many plots as possible, irrespective of missing other input.

jan-kaspar committed:
Re-added horizontal RPs in 210 station - for 2016 and 2017 reprocessing.

jan-kaspar committed:
Fixed id of timing RP, duplicate code removed.

jan-kaspar committed:
Test candidate datasets for RelVals.

forthommel committed:
Renamed keys; new detectors type enum for python configurations

forthommel committed:
Better variable naming, authors list amended

forthommel committed:
Reverted back key naming thanks to hidden boost backward compatibility feature

forthommel committed:
Fixed overriding of last indexing key for calibration dump

jshlee committed:
Merged gem-vfatv3c from repository jshlee with cms-merge-topic

jshlee committed:
remove commented out code

apsallid committed:
finalize before merging with Felice and Marco

apsallid committed:
finalize before merging with Felice and Marco

apsallid committed:
replace cout with LogDebug

apsallid committed:
apply codechecks

apsallid committed:
keeping the densities

apsallid committed:
density in validator

apsallid committed:
adapt multitrackvalidator for HGCal validation

apsallid committed:
remove spaces

apsallid committed:
applying Matti's suggestion

apsallid committed:
Once again add back the Density object that was accidentally removed when trying to syncronize with the recent PR of modernized dictionaries

apsallid committed:
applying first bunch from kevin corrections

apsallid committed:
applying the rest of the corrections from Kevin

apsallid committed:
some tweaks

rappoccio committed:
Adding PF-like GenJet specifics

apsallid committed:
Specify the material budget file used in order to avoid confusion and mistakes

apsallid committed:
removing the assert

apsallid committed:
applying reviewer comments, that is removing couts, correcting classes_def.xml etc.

Commit:
Add 2 algorithms for HGCal to DD4Hep format

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-26158/8751

- This PR adds an extra 16KB to repository

cmsbuild commented:
A new Pull Request was created by  @bsunanda  for master.

It involves the following packages:

DetectorDescription/DDCMS


@civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please review it and eventually sign? Thanks.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

bsunanda commented:
@cmsbuild Please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: ed54d27a5e815ccc054474eba560a26914fe1a99
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26158/33555/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26158/33555/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 3 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 3114829
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3114631
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

jan-kaspar committed:
Test config uses GT with alignment, alignment specification removed from
main reco config.

jan-kaspar committed:
Added labels to LHCInfo data.

jan-kaspar committed:
LHCInfo and optics source specs moved from main reco config to dedicated
cff files.

jan-kaspar committed:
getSplines() renamed to splines().

jan-kaspar committed:
Namespace and struct blocks removed.

jan-kaspar committed:
Always produces output.

jan-kaspar committed:
Added fillDescriptions method.

jan-kaspar committed:
Added labels to LHCInfo data.

jan-kaspar committed:
Removed duplicate configs.

jan-kaspar committed:
Use cff files from CalibPPS/ESProducers.

jan-kaspar committed:
Merge branch 'master' into proton_reco_10_4_X

jan-kaspar committed:
Removed funny charcters inserted by GitHub resolve tool.

guitargeek committed:
Remove RecoParticleFlow/PFProducer/Utils

guitargeek committed:
Removed RecoParticleFlow/PFProducer/interface/GsfElectronEqual.h

guitargeek committed:
Removed RecoParticleFlow/PFProducer/interface/PhotonEqual.h

guitargeek committed:
remove RecoParticleFlow/PFProducer/plugins/EFilter

guitargeek committed:
template instantiation plugins to plugins.cc

Commit:
Revert default tracking classifier back to BDT for Phase1. Added new process modifier trackdnn, that can be used to change the tracking classifier to DNN.

cardinia committed:
Tool for producing DMR trends + doxygen documentation for an older tool used in TkAl OfflineValidation

Commit:
Comply with new CMSSW style rules

Commit:
Use the function in GeantUnits

cmsbuild commented:
The code-checks are being triggered in jenkins.

guitargeek committed:
Use already existing normalizedPhi function

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-26158/8799

- This PR adds an extra 16KB to repository

cmsbuild commented:
Pull request #26158 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

bsunanda commented:
@cmsbuild Please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 33a718d89bb3ac9d26f94b467cbc6a114d82becd
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26158/33626/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26158/33626/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 3114829
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3114631
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

Commit:
Fixed the use of wrong track collection in LowPtTriplet_cff.py trackdnn. Added missing import of trackSelectionLwtnn_cfi to make trackdnn work. Fixed typo in import paths for trackdnn ProcessModifier.

jan-kaspar committed:
classes.h contains headers only.

jan-kaspar committed:
Auto GT used.

jan-kaspar committed:
Common replacement pattern moved to a common cff.py file.

forthommel committed:
Updated the path to calibration example for diamond

apsallid committed:
moving detid related dictionaries to the appropriate place

cardinia committed:
Updated macro after code checks + lumi-per-IOV/run files moved to data directory

Commit:
Update the macros used for calibration of HCAL

jshlee committed:
fix crc issue

Commit:
Add a new scenario D41 with T14+C8+I9+O3

Commit:
Add a new scenario D41 with T14+C8+I9+O3

Commit:
Regenrated file

smuzaffar committed:
cleanup lost dictionaries errors

cvuosalo commented on DetectorDescription/DDCMS/plugins/DDHGCalCell.cc:
```diff
@@ -0,0 +1,286 @@
+  double dx4 = 0.5*dx2;
+  double dy1 = r;
+  double dy2 = 0.5*dy1;
+  double dy3 = 1.5*dy1;
```
The 3.0 in line 76, 2.5 in line 80, and 1.5 here sure look like arbitrary magic numbers. The code would probably be more maintainable if these were constants with explanatory names so, in the future if the HGCal design changes, they can be more easily found and revised.

bsunanda replied:
I do not think this will change in future. I gave the ones which matters

cvuosalo commented on DetectorDescription/DDCMS/plugins/DDHGCalWafer.cc:
```diff
@@ -0,0 +1,93 @@
+				    << " at " << tran << " with " << rotation;
+#endif
+    }
+    ny         += 6;
```
6 also looks like a magic number. Making it a constant with an explanatory name would make the code more understandable.

bsunanda replied:
I added a name

hajohajo committed:
Delete customise_MVAPhase1_cfi.py

francescobrivio committed:
VtxSmearing parameters for Run 3 FlatOptics

jan-kaspar committed:
LogWarning replaced with LogInfo.

apsallid committed:
saving geometry info to avoid hardcoded values

jan-kaspar committed:
PPS protons added to miniAOD.

jan-kaspar committed:
Reverted back to using alignment from local XML file.

jan-kaspar committed:
Added cff file with alignment settings for PPS.

cardinia committed:
Updated DMRtrends (after cmssw checks) + small bugfix for the Zmumu plotting tool

Commit:
Take Carl's advice

cmsbuild commented:
The code-checks are being triggered in jenkins.

bsunanda commented:
@cmsbuild Please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-26158/8859

- This PR adds an extra 16KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #26158 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

cmsbuild commented:
+1
Tested at: 1f605ed47d14d8ff77f71f629ee0375aa4198818
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26158/33700/summary.html

cmsbuild commented:
Comparison job queued.

ianna commented:
@bsunanda - please, check and fix a logic error: Uninitialized argument value: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26158/33700/llvm-analysis/report-889499.html#EndPath

jbsauvan committed:
Update motherboard mappings for v9 and v10 geometries

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26158/33700/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 3114829
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3114631
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

bsunanda commented:
@ianna This is not a logic error. truncCN.size() is supposed to be 3 which is tested above so i could be 0,1,2 and ire with X=0,1,2,3,4 all have 3 values in the array. SO please sign this PR

cvuosalo commented on DetectorDescription/DDCMS/plugins/DDHGCalCell.cc:
```diff
@@ -0,0 +1,288 @@
+				 << " Corners " << cornrCN.size() << ":"
+				 << cornrSensN.size();
+  if ((truncCN.size() != 3) || (extenCN.size() != 3) ||
+      (cornrCN.size() != 6))
```
Shouldn't there be a `return` statement for this `if`? If `truncCN.size()` is greater than 3, the code below will read way out of bounds. Currently, the code just issues a warning and marches along into memory errors.

bsunanda replied:
I am sending an exception

Commit:
Add exception for wrong i/p

cmsbuild commented:
The code-checks are being triggered in jenkins.

bsunanda commented:
@cmsbuild Please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-26158/8863

- This PR adds an extra 16KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #26158 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

cmsbuild commented:
+1
Tested at: 2c1325468803ea9dddb994b154cf1fc49e458314
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26158/33713/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26158/33713/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 3114829
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3114631
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

bmahakud committed:
first commit

jan-kaspar committed:
Conditions settings for proton reco factorised to ctppsProtons_cff.

jan-kaspar committed:
Use ctppsProtons_cff.

kpedro88 committed:
default dataset for D41

kpedro88 committed:
forward D41 to relval_2023

Commit:
Move the warnings to error

cmsbuild commented:
The code-checks are being triggered in jenkins.

bsunanda commented:
@cmsbuild Please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-26158/8867

- This PR adds an extra 20KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #26158 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

cardinia committed:
Fixed bug regarding running the tool from command line

cmsbuild commented:
+1
Tested at: 6a6e0e5f699e7aa019c5ca2f86a1096143604684
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26158/33720/summary.html

cmsbuild commented:
Comparison job queued.

Commit:
Modify mtd geometry by Fabio

rappoccio committed:
Code review changes.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26158/33720/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 3114829
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3114631
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

Commit:
Enable test beam setup to dump geometry for cmsShow

Commit:
Add the changes to hgcalHE

guitargeek committed:
All plugin factories declared in pluginFactories.cc file

bmahakud committed:
function definitions organized

adewit committed:
Alignment/OfflineValidation: Moving to absolute imports at the end of presentation.py

Commit:
Update V10 geometry withinput from Philippe

apsallid committed:
applying comments in CaloParticleSelector and a typo on makeHGCalValidationPlots.py

Dr15Jones committed:
Protect against bad inputs in EnergyUncertaintyElectronSpecific

ASAN reported out of bounds reads in calls to member functions.
We now report if an input value would result in bad array reads.
In addition, moved the arrays from the stack to a compile time
created memory location.

ianna commented:
> @ianna This is not a logic error. truncCN.size() is supposed to be 3 which is tested above so i could be 0,1,2 and ire with X=0,1,2,3,4 all have 3 values in the array. SO please sign this PR

@bsunanda - This is how clang interprets it :-) Please, see clang reasoning here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26158/33720/llvm-analysis/report-c5288b.html#EndPath

perrotta committed:
Fix fillDescriptions for RecoTauPiZeroProducer

bsunanda commented:
@ianna I see the reasoning - but I do not agree with the reasoning. It does not see the exception in front

Commit:
Attempt to avoid clang comments

Commit:
Avoid direct reference to CLHEP

cmsbuild commented:
The code-checks are being triggered in jenkins.

bsunanda commented:
@cmsbuild Please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-26158/8891

- This PR adds an extra 16KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #26158 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @cmsbuild can you please check and sign again.

cmsbuild committed:
Merge pull request #26209 from bsunanda/Run2-alca152

Run2-alca152 Update the macros used for calibration of HCAL

cmsbuild committed:
Merge pull request #26235 from adewit/from-CMSSW_10_6_0_pre2

TkAl AllInOneTool: Moving to absolute imports at the end of presentation.py

cmsbuild commented:
+1
Tested at: cadca32f367d40865d728fe71be0fa39c6096a1e
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26158/33748/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild committed:
Merge pull request #26224 from smuzaffar/106x-lostdef-fix1

[106X] Cleanup lost dictionaries errors

perrotta committed:
Fix TauDiscriminatorProducer

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26158/33748/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 3114829
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3114631
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

cmsbuild committed:
Merge pull request #26212 from bsunanda/Phase2-hgx186

Phase2 hgx186 Define a new scenario D41

cmsbuild committed:
Merge pull request #26195 from cardinia/from-CMSSW_10_5_0_pre2

Tool for producing DMR trends

Commit:
Code review changes: consistency, bug in LowPtTripletStep cut values.

smuzaffar committed:
[10.6.X] Use cmssdt.cern.ch/lxr instead of cmslxr.fnal.gov

perrotta committed:
Some further cleaning

perrotta committed:
just a move...

perrotta committed:
define VPSets defaults where possible

Commit:
Update some of the code for material budget calculation and re-arrange

cmsbuild committed:
Merge pull request #26244 from bsunanda/Run2-hcx190

Run2-hcx190 Avoid direct reference to CLHEP

Commit:
Remove extrusions

cmsbuild committed:
Merge pull request #26237 from bsunanda/Phase2-hgx187

Phase2-hgx187 Update V10 geometry with input from Philippe

Commit:
Take care of hfnose in view of change in algorithm

bsunanda commented:
@ianna @cvuosalo Can you now approve this PR?

Commit:
Update the V0 version

Dr15Jones committed:
Removed unnecessary dictionary struct from DataFormats/HepMCCandidate

ROOT no longer requires explicitly referencing templates in the
classes.h file.

Dr15Jones committed:
Removed unnecessary dictionary struct from DataFormats/TrackingRecHit

ROOT no longer requires explicitly referencing templates in the classes.h file.

Dr15Jones committed:
Removed unnecessary dictionary struct from DataFormats/TrajectorySeed

ROOT no longer requires explicitly referencing templates in the classes.h file.

Dr15Jones committed:
Removed unnecessary dictionary struct from FWCore/MessageLogger

ROOT no longer requires explicitly referencing templates in the classes.h file.

cmsbuild committed:
Merge pull request #26207 from forthommel/ppsDiamTiming_condFormat

[PPS] Conditions format update for diamond timing calibration

cmsbuild committed:
Merge pull request #26223 from MilanoBicocca-pix/VtxSmearingScenario_10_6_X_FlatOptics

Update VtxSmearingScenario for 10_6_X MC production - Flat Optics

cmsbuild committed:
Merge pull request #26190 from guitargeek/PFProducer_cleanup

Some RecoParticleFlow/PFProducer cleanup

cmsbuild committed:
Merge pull request #26254 from perrotta/removeDuplicatePfRecoTauProducer_cfi

Fix configurations and their descriptions in recoTau

jingyucms committed:
add tot plots to HGCal Validation + some minor fixes

jingyucms committed:
add some comments

Commit:
Fixed bug in PixelPairStep HP cut value. Changes to align the code with what it was pre-DNN

cmsbuild committed:
Merge pull request #26262 from Dr15Jones/dictionaryMessageLogger

Removed unnecessary dictionary struct from FWCore/MessageLogger

fabiocos committed:
Squashed commit of the following:

commit 04e265b1fccd65b7459910df529c5782b29d0a4f
Author: Luigi <luigi.calligaris@cern.ch>
Date:   Sat Mar 23 16:46:55 2019 -0300

    Specify the correct CSV filename in the cabling map producer

commit 1e15dc6a9cfae1f7bf2ce229fd9bb7bd68ab2ff0
Author: Luigi <luigi.calligaris@cern.ch>
Date:   Mon Mar 18 14:45:03 2019 -0300

    Bugfix: change addDefault(char const*, ConfigurationDescriptions const&) to add(char const*, ConfigurationDescriptions const&) in plugins constructor, since the former call is not supported

commit cc2a34bac9d1392db71013def7668c58018fb143
Author: Luigi <luigi.calligaris@cern.ch>
Date:   Mon Mar 18 13:51:05 2019 -0300

    Add string in DTCCablingMapTestProducer to automatically generate the CFI file

commit 9c59f77b51ddc3104dfc05bff215ff29cc9b063d
Author: Luigi <luigi.calligaris@cern.ch>
Date:   Mon Mar 18 13:47:44 2019 -0300

    Remove redundant Cfi file for DTCCablingMapProducer

commit a2cf9af5e0c3b773909b86fc0556cf544ce90b35
Author: Luigi <luigi.calligaris@cern.ch>
Date:   Fri Mar 15 18:40:22 2019 -0300

    Clean up Buildfile dependencies, move gbt_id and elink_id ranges from being a hardcoded number to a constexpr at the top of the source in the cabling map producer

commit caa1a0b3d8184c0ece3a00e1c0ae9d6aab958b5d
Author: Luigi <luigi.calligaris@cern.ch>
Date:   Thu Mar 14 18:04:52 2019 -0300

    Fix indentation in DTCCablingMapTestProducer, remove redundant bracket in DTCCablingMapProducer

commit baf86486a92d2930448cffef8392584363254f3c
Author: Luigi <luigi.calligaris@cern.ch>
Date:   Fri Mar 8 20:59:43 2019 -0300

    Move headers.h, which was causing problems to git for unknown reasons

commit 54f1cc9e4aa6996bfab976a0721d0cedb74c5c4e
Author: Luigi <luigi.calligaris@cern.ch>
Date:   Fri Mar 8 20:56:52 2019 -0300

    Rename package directories: part 1 (due to git misbehaviour)

commit 6020571d6aed0a0772775d07b9611c6b1647312b
Author: Luigi <luigi.calligaris@cern.ch>
Date:   Fri Mar 8 19:55:31 2019 -0300

    Remove all references to the old SLHCUpgradeSimulations/Phase2TrackerDTC path for the cabling map producer package

commit a884b8721b0c005988e0dd1b2119208329e95235
Author: Luigi <luigi.calligaris@cern.ch>
Date:   Fri Mar 8 19:15:29 2019 -0300

    Rename SLHCUpgradeSimulations/Phase2TrackerDTCCablingMap to CondTools/SiPhase2Tracker

commit ca7e4c94435542f3e42c32e2ef292f8ceda428b2
Author: Luigi <luigi.calligaris@cern.ch>
Date:   Fri Mar 8 19:01:18 2019 -0300

    Remove three string trimming helper functions, which were not being used

commit 76cdbe6fb2285ef31239a0a4a720b54f751d0b46
Author: Luigi <luigi.calligaris@cern.ch>
Date:   Fri Mar 8 14:30:36 2019 -0300

    Remove references to the geometry in the test psets, as it is not used anymore

commit ec15e77a3b40eba3fb590ad3444b6986f8ae24bf
Author: Luigi <luigi.calligaris@cern.ch>
Date:   Fri Mar 8 14:29:48 2019 -0300

    Remove the move operator in the cabling map insertion, which is not needed

commit 254dd82828e6ebaf5469836d9ae75589076f9d7a
Author: Luigi <luigi.calligaris@cern.ch>
Date:   Thu Feb 28 21:10:47 2019 -0300

    Tracker FE to DTC cabling map, after preliminary review (squashed commits)

mmusich committed:
switch on simulation of the pixel bad components on the FED channel basis

forthommel committed:
Merge branch 'from-CMSSW_10_6_X_2019-03-27-1100' into proton_reco_10_4_X

jingyucms committed:
remove commented-out lines

cmsbuild committed:
Merge pull request #26257 from bsunanda/Phase2-hcx191

Phase2-hcx191 Take care of hfnose in view of change in algorithm

cvuosalo commented:
@ianna: I approve this PR. Please give +1 if you're satisfied with it, too.

cmsbuild committed:
Merge pull request #26099 from apsallid/hgcalvalidator

[HGCal] first version of the HGCalValidator

cmsbuild committed:
Merge pull request #26100 from apsallid/mtvforhgcal

[HGCal] Modify MultiTrackValidator plotting for easier browsing of HGCal plots

cmsbuild committed:
Merge pull request #26125 from rappoccio/pfGenJetSpecifics

Pf gen jet specifics

cmsbuild committed:
Merge pull request #26260 from Dr15Jones/dictionaryTrackingRecHit

Removed unnecessary dictionary struct from DataFormats/TrackingRecHit

cmsbuild committed:
Merge pull request #26261 from Dr15Jones/dictionaryTrajectorySeed

Removed unnecessary dictionary struct from DataFormats/TrajectorySeed

cmsbuild committed:
Merge pull request #26199 from hajohajo/revert-PR21717

Revert default tracking classifier back to BDT for Phase1 (set in PR21717)

ianna commented on DetectorDescription/DDCMS/plugins/DDHGCalWafer.cc:
```diff
@@ -0,0 +1,95 @@
+    int    nx   = 1 - nCellsRow[ir];
+    double ypos = dy*ny;
+    for (int ic=0; ic<nCellsRow[ir]; ++ic) {
+      std::string name(childNames[0]);
```
@bsunanda - please, use string_view to avoid copying a string.

bsunanda replied:
@ianna Can you explain why this is needed - what is the comparison of cost in terms of CPU and memory

bsunanda replied:
I think ns.volume(std::string&) is the relevant method. How string_view can be used rather than string there?

ianna commented on DetectorDescription/DDCMS/plugins/DDHGCalWafer.cc:
```diff
@@ -0,0 +1,95 @@
+      nx         += incAlongX;
+      dd4hep::Position tran(xpos, ypos, 0);
+      int copy = cellType*1000+kount;
+      mother.placeVolume(ns.volume(name), copy, 
```
@bsunanda - here you look-up a Volume by name. Which algorithms creates it, given that you've just constructed the name?

bsunanda replied:
It is not created by algorithm. It is produced in the xml file - so it can be used only in the link of xml files. It is translation of the algorithm with the same name in Geometry/HGCalCommonData and is used for V6, V7, V8 version of HGCal and all TB applications

ianna commented:
@bsunanda - it would be good to have a test to check that the algorithms work correctly.

cmsbuild committed:
Merge pull request #26259 from Dr15Jones/dictionaryHepMCCandidate

Removed unnecessary dictionary struct from DataFormats/HepMCCandidate

cmsbuild committed:
Merge pull request #26268 from fabiocos/fc-squash26029

Phase2 tracker dtc cabling map (squashed version)

ianna commented:
@cvuosalo IMHO we need to encourage developers to provide tests for the code they submit.

davidlange6 committed:
move to pybind11 parameteset reader as default. fix some buildfiles

fabiocos committed:
Add python3 compatible print management

zhenhu committed:
update job splitting for 2018 HI data workflows

Dr15Jones commented:
> @cvuosalo IMHO we need to encourage developers to provide tests for the code they submit

I definitely agree with that.

cmsbuild committed:
Merge pull request #26250 from cms-sw/smuzaffar-patch-7

[10.6.X] Use cmssdt.cern.ch/lxr instead of cmslxr.fnal.gov

davidlange6 committed:
work around shortcoming I need to fix

bsunanda commented:
@ianna Why don't you prepare a test which can compare two geometry trees produced from DD4HEP and DetectorDescriptionLanguage? Then we can then compare the solutions from 2 generations - that is the best test

fabiocos committed:
Add modifier for premixing phase2

cmsbuild committed:
Merge pull request #26198 from dildick/from-CMSSW_10_6_0_pre2-CSCTriggerPrimitives-CMSSW-style-rules

Make L1Trigger/CSCTriggerPrimitives comply with new CMSSW style rules.

cvuosalo commented:
@ianna: Yes, I agree, unit tests should be included.

ianna commented:
> @ianna Why don't you prepare a test which can compare two geometry trees produced from DD4HEP and DetectorDescriptionLanguage? Then we can then compare the solutions from 2 generations - that is the best test

@bsunanda - please, see a validation example here: old vs new DT geometry is compared https://github.com/cms-sw/cmssw/pull/26231

However, we are not at the point when we can compare all the other sub-detectors. The first step is to port the algorithms. In order to make sure that the algorithm does a correct job, e.g. produces a set of shapes as expected, I would suggest to have a unit-test per algorithm. It can be as simple as: https://cmssdt.cern.ch/lxr/source/DetectorDescription/DDCMS/test/python/testDDAngularAlgorithm.py

cmsbuild committed:
Merge pull request #26279 from fabiocos/fc-fixHGCalValid106X

Add python3 compatible print management in Validation/HGCalValidation

ianna commented:
@bsunanda - Please, take my branch  **ianna:dd4hep-hgcalcell-algorithm-test** where I've implemented a test for one of the algorithms: DDHGCalCell
```
cmsRun DetectorDescription/DDCMS/test/python/testDDHGCalCellAlgorithm.py
```
it runs fine, but if you check what it produces:
```
cmsShow -c geo.fwc --tgeo-name=TestDDHGCalCell --sim-geom-file cmsDD4HepGeom.root
```

Commit:
Make changes to do geometry validation of V10

Commit:
One missed file

cmsbuild committed:
Merge pull request #26287 from fabiocos/fc-fixValidHGCal106X

Fix HGCalValidator configuration for premixing

tocheng committed:
Update phase-1 MC GT to activate Simulation of the Pixel Bad Components on the FED channel basis

cmsbuild committed:
Merge pull request #26278 from davidlange6/pybind11_190328

move to pybind11 parameteset reader as default. fix some buildfiles

ianna commented:
@bsunanda - BTW, I do not see the cell produced with the algorithm. It looks like it's not added to the paren by this algorithm.

bsunanda commented:
@ianna The cells are defined in the xml file through solid-section, logical volume section and then positioned through the algorithm inside the wafer which are also created in the same way in the xml file - please see HGCalCommonData/data/hgcalwafer/v8/hgcalwafer.xml

bsunanda commented:
@ianna I merged your branch with mine but did not see the example. All files in test/python are
UnitsCheck.py              testDDDetectorESProducer.py  testMuonGeometry.py
dump.py                    testDDPseudoTrapShapes.py    testMuonNumbering.py
dumpDDShapes.py            testDDSpecPars.py            testNavigateGeometry.py
dumpMFGeometry.py          testDDSpecParsFilter.py      testShapes.py
dumpMuonGeometry.py        testDDVectors.py             testTGeoIterator.py
testDDAngularAlgorithm.py  testMFGeometry.py

cmsbuild committed:
Merge pull request #25158 from jshlee/gem-vfatv3

GEM unpacker for  VFat V3

cmsbuild committed:
Merge pull request #25495 from CTPPS/proton_reco_10_4_X

PPS proton reconstruction

cmsbuild committed:
Merge pull request #26264 from jingyucms/jingyu-DevBranch

HGCal Validation Plots

cmsbuild committed:
Merge pull request #26241 from Dr15Jones/fixEnergyUncertaintyElectronSpecific

Protect against bad inputs in EnergyUncertaintyElectronSpecific

ianna commented:
> @ianna I merged your branch with mine but did not see the example. All files in test/python are
> UnitsCheck.py testDDDetectorESProducer.py testMuonGeometry.py
> dump.py testDDPseudoTrapShapes.py testMuonNumbering.py
> dumpDDShapes.py testDDSpecPars.py testNavigateGeometry.py
> dumpMFGeometry.py testDDSpecParsFilter.py testShapes.py
> dumpMuonGeometry.py testDDVectors.py testTGeoIterator.py
> testDDAngularAlgorithm.py testMFGeometry.py

@bsunanda - sorry, my bad. Please, try again.

cmsbuild committed:
Merge pull request #26233 from bmahakud/PUPPICombinedIsolationInMuonSelector

Adding PUPPI combined isolation Working Points in miniAOD muon selector

cmsbuild committed:
Merge pull request #26275 from mmusich/activate_stuckTBMSimulation_106X

Activate Simulation of the Pixel Bad Components on the FED channel basis

cmsbuild committed:
Merge pull request #26290 from bsunanda/Phase2-hgx188

Phase2-hgx188 Make changes to do geometry validation of V10

cmsbuild committed:
Merge pull request #26234 from bsunanda/Run2-TB33

Run2-TB33 Enable test beam setup to dump geometry for cmsShow

cmsbuild committed:
Merge pull request #26280 from zhenhu/updateJobSpiltingForHI

update job splitting for 2018 HI data workflows

cmsbuild committed:
Merge pull request #26256 from bsunanda/Run2-TB34

Run2-TB34 Update some of the code for material budget calculation and re-arrange

Commit:
Merged Phase2-hgx185 from repository bsunanda with cms-merge-topic

Commit:
Adding the testing tool

cmsbuild commented:
The code-checks are being triggered in jenkins.

bsunanda commented:
I am closing this PR. Best if Ianna can put in the modified ianna:dd4hep-ddhgcalcell-algo-test-v2 branch in PR or I can take the relevant part of the code in a new PR

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-26158/9033

- This PR adds an extra 28KB to repository

- Found files with invalid states:
  - Alignment/OfflineValidation/interface/lumiperrun2017.txt:
    - Added: 840e6e70086f643d8a55f7c5cad5b3fe2ac9f1a7
    - Deleted: 59ed6fc1e699c8afb1e013a9b122e8181199dbcc
  - Alignment/OfflineValidation/interface/lumiperrun2018.txt:
    - Added: 840e6e70086f643d8a55f7c5cad5b3fe2ac9f1a7
    - Deleted: 59ed6fc1e699c8afb1e013a9b122e8181199dbcc
  - Alignment/OfflineValidation/interface/lumiperIOV2017.txt:
    - Added: 840e6e70086f643d8a55f7c5cad5b3fe2ac9f1a7
    - Deleted: 59ed6fc1e699c8afb1e013a9b122e8181199dbcc
  - Alignment/OfflineValidation/interface/lumiperIOV2018.txt:
    - Added: 840e6e70086f643d8a55f7c5cad5b3fe2ac9f1a7
    - Deleted: 59ed6fc1e699c8afb1e013a9b122e8181199dbcc

cmsbuild commented:
Pull request #26158 was updated. @andrius-k, @schneiml, @ianna, @kpedro88, @rekovic, @fioriNTU, @alberto-sanchez, @pohsun, @santocch, @perrotta, @civanch, @zhenhu, @cmsbuild, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @prebello, @tocheng, @jfernan2, @mdhildreth, @slava77, @ggovi, @qliphy, @fabiocos, @efeyazgan, @kmaeshima, @perrozzi, @pgunnell, @franzoni can you please check and sign again.
